LICENSE

```diff
@@ -1,31 +1,4 @@
-Copyright (c) 2018 King's College London
-created by the Software Development Team <http://soft-dev.org/>
-
-The Universal Permissive License (UPL), Version 1.0
-
-Subject to the condition set forth below, permission is hereby granted to any person obtaining a
-copy of this software, associated documentation and/or data (collectively the "Software"), free
-of charge and under any and all copyright rights in the Software, and any and all patent rights
-owned or freely licensable by each licensor hereunder covering either (i) the unmodified
-Software as contributed to or provided by such licensor, or (ii) the Larger Works (as defined
-below), to deal in both
-
-(a) the Software, and
-(b) any piece of software and/or hardware listed in the lrgrwrks.txt file
-if one is included with the Software (each a "Larger Work" to which the Software is contributed
-by such licensors),
-
-without restriction, including without limitation the rights to copy, create derivative works
-of, display, perform, and distribute the Software and make, use, sell, offer for sale, import,
-export, have made, and have sold the Software and the Larger Work(s), and to sublicense the
-foregoing rights on either these or other terms.
-
-This license is subject to the following condition: The above copyright notice and either this
-complete permission notice or at a minimum a reference to the UPL must be included in all copies
-or substantial portions of the Software.
-
-THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
-BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
-NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
-DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
-OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+Except as otherwise noted, this project is licensed under the Apache License,
```

Comment:
Wouldn't it be an idea to show the licenses in full in this file? Web pages disappear.

Reply:
I've stolen this from the way Rust does things. The Apache license is so long I wasn't sure if I wanted to use up so much space. But I can see your point. Let me have a quick think about it.

Reply:
It's only in one place, so I have no objection to including it in full.

Reply:
Aha, I think I now see why Rust doesn't include the text. Consider the standard MIT license text:

```
Copyright <YEAR> <COPYRIGHT HOLDER>
...
The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
```

If we expand this in the `LICENSE` file, it suggests everything in the crate has a single copyright holder, which is incorrect. But if we leave this on a per-file basis, everything makes sense. So simply deferring people to the website and individual files is quite a clever hack to avoid this problem.

In an ideal world, I think I'd probably drop the `LICENSE` file entirely, but I do appreciate that people often want a quick overview of the licence without wading through every source file.

Thoughts?